sdk/bare/common/sys/Logging-Binary-Format.md

```diff
@@ -0,0 +1,40 @@
+# Logging Binary Format
+
+## Overview
+
+The AMDC can dump the logged variables to the UART serial console using either text-based formatting or the binary
+format. This document describes the binary formatting which appears on the UART interface.
+
+## Dumping Variables
+
+Each logged variable must be dumped one at a time (i.e. if the user has 10 slots, each must be dumped one by one).
+
+To start a dump of the Nth variable slot: `log dump bin N`
+
+## Binary Format
+
+All data is little-endian.
+
+### Magic Header (16 bytes)
+
+0x12345678 (repeated four times)
```

Comment:
What is the purpose of having the header field repeat four times?

There isn't much of a disadvantage to having such a long field (it only adds a minimal number of additional bytes as overhead).

Reply:
I originally only had 32 bits for the header field, however, I got concerned that this magic sequence could appear within the data stream if enough samples were sent (i.e. in the float data type bits). Therefore, to improve the odds, I replicated it four times. Still not fool-proof, but the odds of 128 bits appearing in order is VERY slim. And, its only 3 extra words, so very little overhead.